client/app/vps/upgrade/vps-upgrade.controller.js

```diff
@@ -10,34 +10,33 @@ class VpsUpgradeCtrl {
         this.Vps = VpsService;
 
         this.loaders = {
-            init: false
+            step1: false,
+            step2: false
         };
 
-        this.step = 0;
+        this.completed = {
+            step1: false,
+            step2: false
+        };
         this.order = null;
         this.selectedModel = {};
         this.upgradesList = null;
     }
 
     $onInit () {
         this.previousState = this.CloudNavigation.getPreviousState();
-        this.loaders.init = true;
-        this.loadNextStep();
-    }
-
-    loadNextStep (step) {
-        this.step = step || this.step + 1;
     }
 
     getCurrentModel () {
         return _.find(this.upgradesList, upgrade => upgrade.isCurrentModel === true);
     }
 
     loadUpgradesList () {
-        this.loaders.step1 = true;
         if (!this.upgradesList) {
+            this.loaders.step1 = true;
             return this.Vps.upgradesList(this.serviceName).then(data => {
                 this.upgradesList = data.results;
+                this.selectedModel.model = this.getCurrentModel().model;
                 return data;
             }).catch(err => {
```

Comment:
Consider to do a return $q.reject(err) or $log.error(err) if it's a technical error

Reply:
IMHO It's strange and not standard to do a this.previousState.go() on error

Reply:
We don't want people to stay in upgrade page, if we cannot load the upgrade list.